Doc test data instructions (#45)
ghiggi/gpm_api: CONTRIBUTING.rst

   -  If applicable, ensure that your fix includes appropriate tests. Adding tests for your bug fix helps prevent future regressions and maintains the stability of the software.

Contributing to test data
~~~~~~~~~~~~~~~~~~~~~~~~~

If your changes include updating test data, you may work with the ``gpm/tests/data`` directory as you would with any other git repository.
You need to ask the maintainers to become a contributor on the `gpm_api_test_data <https://github.com/ghiggi/gpm_api_test_data>`_ repository to create a branch and a Pull Request.

> The API repository keeps track of the currently checked-out commit of the test-data repository. When the checked-out commit changes, you can register this change in the API repository by running

Reviewer: ```suggestion
The GPM-API repository keeps track of the currently checked-out commit of the test-data repository. When the checked-out commit changes, you can register this change in the GPM-API repository by running
```

.. code-block:: bash

    git add gpm/tests/data

and committing.

To submit your contribution that involves modifying test data, please follow this procedure.

> (A: API repository, B: test-data repository)

Reviewer: ```suggestion
(A: GPM-API repository, B: test-data repository)
```

> 1. Make a *feature branch* for B

Reviewer: Should we specify maybe that to make a feature branch the contributor should first go within the gpm/tests/data directory?

2. Have A point to the *feature branch* of B
3. Make two PRs (for A and B) and get both accepted

> 4. Have the B’s PR merged into the *main branch*

Reviewer: ```suggestion
4. Have the B’s PR merged into the B *main branch*
```

> 5. Update A to point to B’s *main branch* instead of *feature branch*

Reviewer: ```suggestion
5. Update your A branch to point to B’s new *main branch* (instead of the old *feature branch*)
```

   -  If applicable, ensure that your fix includes appropriate tests. Adding tests for your bug fix helps prevent future regressions and maintains the stability of the software.

Contributing to test data
~~~~~~~~~~~~~~~~~~~~~~~~~

> If your changes include updating test data, you may work with the ``gpm/tests/data/`` directory as you would with any other git repository.

Reviewer: ```suggestion
If your changes modify the structure of the GPM-API xarray Dataset, you will likely need to update the test data in the ``gpm/tests/data/`` directory.  
```

> You need to ask the maintainers to become a contributor on the `gpm_api_test_data <https://github.com/ghiggi/gpm_api_test_data>`_ repository to create a branch and a Pull Request.

Reviewer: ```suggestion
To do this, you need to ask the maintainers to become a contributor on the `gpm_api_test_data <https://github.com/ghiggi/gpm_api_test_data>`_ repository in order that you can create a branch and a Pull Request.
Then you will able to modify the ``gpm/tests/data/`` directory/repository as you would with any other git repository.
```

Reviewer: Is it necessary that someone ask to become a contributor? He can't just fork and PR?

Author: @ghiggi well if you make a fork of the test-data repo, you need to update the submodule url in the gpm-api repo (and restore it before merging the pr), which imo complexify even more the process.

Git LFS would have been an elegant alternative solution but bandwidth quotas are absurly low :(

Reviewer: Ah right. Didn't thought about that ...  
Just merge the suggestion and I merge the PR @sphamba ;)

The GPM-API repository keeps track of the currently checked-out commit of the test-data repository. When the checked-out commit changes, you can register this change in the GPM-API repository by running

.. code-block:: bash

    git add gpm/tests/data

and committing.

To submit your contribution that involves modifying test data, please follow this procedure.

(A: GPM-API repository, B: test-data repository)

1. Make a *feature branch* for B

.. code-block:: bash

    cd gpm/tests/data
    # Inside this directory, following git commands will apply to B
    git checkout -b my-feature-branch
    ...

2. Have A point to the *feature branch* of B

.. code-block:: bash

    # From the root of the GPM-API repository
    git add gpm/tests/data
    git commit
    ...

3. Make two PRs (for A and B) and get both accepted
4. Have the B’s PR merged into the B's *main branch*
5. Update A to point to B’s updated *main branch* (instead of the old *feature branch*)

.. code-block:: bash

    # Checkout the main branch of the test-data repository
    cd gpm/tests/data
    git checkout main
    git pull

>     cd ../../..

Reviewer: Maybe not necessary? /.../.../

Author: `cd ...` does not work in bash, it does in zsh
